### DISCUSSION TITLE
CMSIS SVD schema and non-secure alias

### DISCUSSION THREAD

**Discussion by renesas-jbrabender** [ARM-software/CMSIS_5#1640] [Q&A]
What is the preferred way of describing peripherals mapped to both secure and non-secure alias space?
In rzone schema there is concept of "physical" element, but I am not seeing anything equivalent in svd schema
- Is the expectation just to use a "derivedFrom" and give it another name?

**Comment by JonatanAntoni** [comment]
Hi @renesas-jbrabender,

Please note that the SVD Spec moved into Open-CMSIS-Pack already a while ago. The new repository location is at https://github.com/Open-CMSIS-Pack/svd-spec.

SVD Spec has not been extended to reflect TrustZone properties to that extend, yet. You could use derived from to duplicate the peripheral at a second base address. Adding name suffixes like "(S)" and "(NS)" could help to distinguish. The user needs to know that both peripherals "aliases" actually refer to a single one. 

The SVD spec has the element `protection` (see https://open-cmsis-pack.github.io/svd-spec/main/elem_peripherals.html#elem_peripheral) to let the debugger know which permission is required.

Cheers,
Jonatan
